### PR TITLE
Formatter fix

### PR DESCRIPTION
**Pull request by barry-dutton-ck**
Fix an issue caused by a field which is present, but undefined

### CODE REVIEW

**Review by AlecHFerguson — CHANGES_REQUESTED** [review]
Tests? Or is this functionality covered by the tests you added in #40 ?

**Comment by barry-dutton-ck** [comment]
@AlecHFerguson One of these days I am going to remember to write tests before PR'ing!

Fixed and also addressed lint issues:
```
47 tests complete
Test duration: 104 ms
Assertions count: 125 (verbosity: 2.66)
No global variable leaks detected
Coverage: 95.30% (21/447)
lib/formatters.js missing coverage on line(s): 10, 18, 19, 20, 110
lib/good-influx-udp.js missing coverage on line(s): 16
lib/line-protocol.js missing coverage on line(s): 27, 28, 29, 41, 42, 48, 50, 106, 108, 113, 122, 134
lib/ops-format.js missing coverage on line(s): 65, 66, 84
Linting results: No issues
```

File: lib/formatters.js [hunk 1]
```diff
@@ -92,9 +92,12 @@ const flatten = (data, prefix = 'data') => {
         });
         return returnVal;
     }
-    return {
-        [`${prefix}`]: formatVal(data)
-    };
+    else if (data || (data === false)) {
```
Comment (elreimundo) [inline lib/formatters.js]: As a nit, `else` is unnecessary after a `return`. Also, this feels wrong; what if `data` is `0` or `NaN`? Seems like you really want

    if (data == null) {
        return {};
    }
    return {
        [`${prefix}`]: formatVal(data)
    }

Comment (barry-dutton-ck) [reply lib/formatters.js]: Good catch. I changed this to specifically catch null and undefined.

**Review by riley-stroud-ck — APPROVED** [review]

**Review by AlecHFerguson — APPROVED** [review]

**Review by Vilius-Zaikauskas-CK — APPROVED** [review]

**Pull request merged**